File: projects/order-creation-queries/core/src/main/java/com/robinfood/core/mappers/OrderDetailDiscountMappers.java
package com.robinfood.core.mappers;

import com.robinfood.core.dtos.OrderDetailDiscountDTO;
import com.robinfood.core.entities.OrderDetailDiscountEntity;

public final class OrderDetailDiscountMappers {

    private OrderDetailDiscountMappers() {
        throw new IllegalStateException("Utility class");
    }

    public static OrderDetailDiscountDTO toOrderDetailDiscountDTO(
            OrderDetailDiscountEntity orderDetailDiscountEntity
    ) {
        return new OrderDetailDiscountDTO(
                orderDetailDiscountEntity.getId(),
                orderDetailDiscountEntity.getTypeId(),
                orderDetailDiscountEntity.getValue()
        );
    }
}

File: projects/posv2/core/src/main/kotlin/com/robinfood/core/mappers/OrderDetailDiscountMappers.kt
package com.robinfood.core.mappers

import com.robinfood.core.dtos.OrderDetailDiscountDTO
import com.robinfood.core.entities.OrderDetailDiscountEntity
import java.math.BigDecimal

fun OrderDetailDiscountEntity.toOrderDetailDiscountDTO(): OrderDetailDiscountDTO? {
    return if (id == null || typeId == null) {
        null
    } else {
        return OrderDetailDiscountDTO(
                id = id,
                typeId = typeId,
                value = value ?: BigDecimal.ZERO
        )
    }
}